ui/src/app/components/workflows/workflow/workflow.component.ts

```diff
@@ -68,6 +76,10 @@ export class WorkflowComponent implements OnInit, OnDestroy {
     this.isJobsAccordionHidden = !this.isJobsAccordionHidden;
   }
 
+  cancelWorkflow() {
+    !!this.previousRouteService.getPreviousUrl() ? this.location.back() : this.router.navigateByUrl(absoluteRoutes.WORKFLOWS_HOME);
```

Comment:
When the user clears the history while he is on the workflow page, does it work then? Why can't we just call `navigateByUrl(getPreviousUrl())`?

Reply:
It will work because history is saved in our service (even if that is empy it will go to . WORKFLOWS_HOME page).
Because history can be empty and you dont wont to redirect user to blank browser page but to something in your app (in this case it is WORKFLOWS_HOME page) .

Reply:
After discussion, yes you are right. Fixed.